src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -112,7 +114,28 @@ protected set
                 _handshakeRequest = new HandshakeRequestMessage(serviceProtocol.Version, (int)connectionType, (int)migrationLevel);
             }
 
-            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            var currentNamespace = typeof(ServiceConnectionBase).Namespace;
+            Logger = loggerFactory.CreateLogger($"{currentNamespace}.ServiceConnection"); // keep original category name
+            /*
+             * Put all messaging log to Info level isn't appropriate, to Debug level, there are too many ping logs. So plan to put into a new category'
+             * Let CX filter messaging log by appsettings.json's logging filter: https://github.com/Azure/azure-signalr/blob/dev/docs/howto-tsg.md#add_logs_server_aspnetcore
+             * {
+                    "Logging": {
+                        "LogLevel": {
+                            ...
+                            "Microsoft.Azure.SignalR.Messaging": "Debug",

```

Comment:
if customer writes `"Microsoft.Azure.SignalR": "Debug"`, it will also include all the message logs. Is that expected?